handlers.go

```diff
@@ -310,17 +317,23 @@ func (b *BouncerHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	// We want to return ESR115 when... the product is for Firefox
 	shouldReturnESR115 := strings.HasPrefix(reqParams.Product, "firefox-") &&
-		// and the product is _not_ an MSI build
-		!strings.Contains(reqParams.Product, "-msi") &&
 		// and the product is _not_ a partial or complete update (MAR files)
 		!strings.Contains(reqParams.Product, "-partial") &&
 		!strings.Contains(reqParams.Product, "-complete") &&
-		// and the OS param specifies windows
-		strings.HasPrefix(reqParams.OS, "win") &&
-		// and the User-Agent says it's a Windows 7/8/8.1 client
-		isUserAgentOnlyCompatibleWithESR115(req.UserAgent()) &&
 		// and the request doesn't come from mozilla.org
-		!hasMozorgReferrer(reqParams.Referer)
+		!hasMozorgReferrer(reqParams.Referer) && (
+	// AND the OS param specifies windows
+	(strings.HasPrefix(reqParams.OS, "win") &&
+		// and the product is _not_ an MSI build
+		!strings.Contains(reqParams.Product, "-msi") &&
+		// and the User-Agent says it's a Windows 7/8/8.1
+		isWindowsUserAgentOnlyCompatibleWithESR115(req.UserAgent())) ||
+		// OR the OS param specifies macOS
+		(reqParams.OS == "osx" &&
+			// and the product is _not_ a (macOS) pkg build
+			!strings.Contains(reqParams.Product, "-pkg") &&
+			// and the User-Agent says it's a macOS 10.12/10.13/10.14 client
+			isMacOSUserAgentOnlyCompatibleWithESR115(req.UserAgent())))
```

Comment:
The indentation here seems off, it'd be nicer if the win and osx cases were symmetric.  Maybe use intermediate variables instead of the giant boolean formula of doom?

Reply:
yeah, I was thinking about that too, quite sad that golang doesn't allow more freedom in how to indent the code/condition...